docs/protocol/on-chain/security.mdx

```diff
@@ -0,0 +1,286 @@
+3. **Monitor events**: Watch the <Tooltip content="The underlying distributed ledger technology (e.g., Ethereum).">blockchain</Tooltip> for <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> events (Challenged, Closed)
+4. **Respond promptly**: React to challenges within the challenge period
+5. **Verify adjudicators**: Only use <Tooltip content="A smart contract that validates state transitions according to application-specific rules.">adjudicator</Tooltip> contracts from trusted sources
+
+:::danger Critical Rule
+**NEVER sign two different states with the same version number.** This creates ambiguity about the true latest state and can lead to disputes.
+:::
+
+### For Implementers
+
+**Implementation Requirements**:
+
+1. **Validate thoroughly**: Check all inputs before submitting transactions
+2. **Use adjudicators wisely**: Leverage <Tooltip content="A smart contract that validates state transitions according to application-specific rules.">adjudicators</Tooltip> to enforce application rules
+3. **Set appropriate challenge periods**: Balance security needs with user experience
+4. **Implement proper key management**: Secure storage for <Tooltip content="An entity (identified by a wallet address) that is part of a channel.">participant</Tooltip> private keys
+5. **Log state transitions**: Maintain audit trail of all <Tooltip content="A snapshot of the channel at a point in time, including fund allocations and application-specific data.">state</Tooltip> updates
+
+**Sample Validation Checklist**:
+
+```markdown
+Before submitting state on-chain:
+☐ Verify all required signatures present
+☐ Verify signatures are valid for expected participants
+☐ Verify state version is sequential
+☐ Verify allocations sum correctly
+☐ Verify magic numbers (CHANOPEN/CHANCLOSE) if applicable
+☐ Verify channelId matches expected value
+☐ Test with small amounts first
+```
+
+### For Adjudicator Developers
+
+**Critical Requirements**:
+
+1. **Implement strict version comparison**: Ensure newer states always supersede older ones
+2. **Validate state transitions**: Enforce application-specific rules correctly
+3. **Optimize for gas efficiency**: Validation happens on-chain during disputes
+4. **Consider edge cases**: Handle all possible state transition scenarios
+5. **Audit thoroughly**: Security review before deployment is essential
+
+:::warning Adjudicator Responsibility
+<Tooltip content="A smart contract that validates state transitions according to application-specific rules.">Adjudicators</Tooltip> are critical to <Tooltip content="A secure communication pathway between participants that locks funds in an on-chain smart contract while enabling off-chain state updates.">channel</Tooltip> security. A flawed adjudicator can undermine the entire channel's safety guarantees.
+:::
+
+**Adjudicator Development Checklist**:
```

Comment:
I would remove this checklist as it encourages readers to develop their own Adjudicator contract. As we want the opposite, let's add a callout noting that:

Adjudicator is an incredibly important part of the Nitrolite protocol. Yellow Network is built on top of a specific adjudicator, which if changed, will render interoperability and security guarantees impossible. Before starting implementing your own Adjudicator, please be sure to advise the Nitrolite developer team, so that your work is not left out.